src/internal/converters/ics/ics.go

```diff
@@ -0,0 +1,444 @@
+package ics
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/alcionai/clues"
+	ics "github.com/arran4/golang-ical"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/common/str"
+	"github.com/alcionai/corso/src/pkg/dttm"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+)
+
+// This package is used to convert json response from graph to ics
+// Ref: https://icalendar.org/
+// Ref: https://www.rfc-editor.org/rfc/rfc5545
+// Ref: https://learn.microsoft.com/en-us/graph/api/resources/event?view=graph-rest-1.0
+
+// TODO: Items not handled
+// locations (different from location)
+// exceptions and modifications
+
+// Field in the backed up data that we cannot handle
+// allowNewTimeProposals, hideAttendees, importance, isOnlineMeeting,
+// isOrganizer, isReminderOn, onlineMeeting, onlineMeetingProvider,
+// onlineMeetingUrl, originalEndTimeZone, originalStart,
+// originalStartTimeZone, reminderMinutesBeforeStart, responseRequested,
+// responseStatus, sensitivity
+
+func keyValues(key, value string) *ics.KeyValues {
+	return &ics.KeyValues{
+		Key:   key,
+		Value: []string{value},
+	}
+}
+
+func getLocationString(location models.Locationable) string {
+	if location == nil {
+		return ""
+	}
+
+	loc := ""
+	dn := ptr.Val(location.GetDisplayName())
+	segments := []string{dn}
+
+	// TODO: Handle different location types
+	addr := location.GetAddress()
+	if addr != nil {
+		street := ptr.Val(addr.GetStreet())
+		city := ptr.Val(addr.GetCity())
+		state := ptr.Val(addr.GetState())
+		country := ptr.Val(addr.GetCountryOrRegion())
+		postal := ptr.Val(addr.GetPostalCode())
+
+		segments = append(segments, street, city, state, country, postal)
+	}
+
+	nonEmpty := []string{}
+
+	for _, seg := range segments {
+		if len(seg) > 0 {
+			nonEmpty = append(nonEmpty, seg)
+		}
+	}
+
+	if len(nonEmpty) > 0 {
+		loc = strings.Join(nonEmpty, ", ")
+	}
+
+	return loc
+}
+
+func getUTCTime(ts, tz string) (time.Time, error) {
+	// Timezone is always converted to UTC. We should do
+	// this everywhere. This is the easiest way to ensure we
+	// have the correct time everywhere and is the same(need
+	// to same according to spec).
+	it, err := dttm.ParseTime(ts)
+	if err != nil {
+		return time.Now(), clues.Wrap(err, "parsing time")
+	}
+
+	timezone, ok := GraphTimeZoneToTZ[tz]
+	if !ok {
```

Comment:
you may need to check if Outlook allows creating events without a specified timezone (i.e. it just occurs at the same time of day in the local timezone no matter where the user is). The iCal spec seems to allow for this (called a "floating time" or something) but the code here seems like it may end up returning an error and skipping the item if Outlook allows creating an event like that

Reply:
Does not look like from docs. When trying to create one without a timezone, I get the following error.

```json
{
  "error": {
    "code": "TimeZoneNotSupportedException",
    "message": "A valid TimeZone value must be specified. The following TimeZone value is not supported: ''."
  }
}
```

Reply:
great, thanks for checking that!